plugins/org.yakindu.sct.model.stext.ui/src/org/yakindu/sct/model/stext/ui/quickfix/EventTriggerCreationCommand.xtend

```diff
@@ -0,0 +1,92 @@
+/** 
+ * Copyright (c) 2019 itemis AG and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ * Contributors:
+ * itemis AG - initial API and implementation
+ */
+package org.yakindu.sct.model.stext.ui.quickfix
+
+import java.util.Collections
+import org.eclipse.core.runtime.IAdaptable
+import org.eclipse.core.runtime.IProgressMonitor
+import org.eclipse.emf.ecore.EObject
+import org.eclipse.emf.ecore.util.EcoreUtil
+import org.eclipse.emf.transaction.util.TransactionUtil
+import org.eclipse.gmf.runtime.common.core.command.CommandResult
+import org.eclipse.gmf.runtime.emf.commands.core.command.AbstractTransactionalCommand
+import org.eclipse.xtext.validation.Issue
+import org.yakindu.base.types.Direction
+import org.yakindu.base.types.TypesFactory
+import org.yakindu.sct.model.sgraph.Statechart
+import org.yakindu.sct.model.stext.stext.InterfaceScope
+import org.yakindu.sct.model.stext.stext.StextFactory
+
+class EventTriggerCreationCommand extends AbstractTransactionalCommand {
+
+	protected extension StextFactory factory = StextFactory.eINSTANCE
+	protected extension TypesFactory typesFactory = TypesFactory.eINSTANCE
+
+	protected EObject contextElement;
+	protected Issue issue;
+	Direction desiredDirection
+
+	new(EObject element, Issue issue, Direction desiredDirection) {
+		super(TransactionUtil.getEditingDomain(element), "Create missing event", Collections.emptyList);
+		if (issue.data.length < 2) {
+			throw new IllegalArgumentException(
+				"data[0] is expected to contain an URI from a embedded resource to the statechart, " +
+					"data[1] is expected to contain the semantic name of the invalid EObject"
```

Comment:
`issue.data is expected to have two entries: (1) an URI to the statechart resource and (2) the name of the unresolved reference.`